src/error_handling.c

```diff
@@ -167,6 +167,24 @@ rcutils_set_error_state(
   __rcutils_reset_error_string(&__rcutils_error_string, __rcutils_error_state->allocator);
 }
 
+bool 
+rcutils_set_formatted_error(
+  char * formatted_msg,
+  rcutils_allocator_t allocator)
+{
+  if (!formatted_msg) {
+    RCUTILS_SET_ERROR_MSG("No invalid memory allocated to format string", allocator);
+    return false;
+  }
+
+  RCUTILS_SET_ERROR_MSG(formatted_msg, allocator);
```

Comment:
Because you implemented this as a function rather than a maco, (and then call the macro here) the error messages that use this will always get a file and line number here, rather than where it was called from. This isn't correct imo, so this would need to be implemented as a macro instead.